lib/rails_core_extensions/translations.rb

```diff
@@ -10,24 +10,20 @@ def translate(key, options = {})
       end
 
       def translation_key
-        (base_activerecord_class || base_ruby_class).name.tableize.singularize
+        base_translation_class.name.tableize.singularize
       end
 
-      def base_activerecord_class
-        return unless defined?(ActiveRecord) && is_a?(ActiveRecord::Base)
+      def base_translation_class
         klass = self
-        while klass.superclass != ActiveRecord::Base
+        while !base_classes.include? klass.superclass
           klass = klass.superclass
         end
         klass
       end
 
-      def base_ruby_class
-        klass = self
-        while klass.superclass != Object
-          klass = klass.superclass
-        end
-        klass
+      def base_classes
```

Comment:
You could memoize this method, or use an block based loop in base_translation_class
```ruby
def base_translation_class
  ancestors.find { |a| base_classes.include? a } # Note: ancestors include self.class
end
```